Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1107,6 +1107,9 @@ void RenderBlockFlow::layoutBlockChild(RenderBox& child, MarginInfo& marginInfo,
             layoutState->pushBlockStartTrimming(false);
         }
     }
+
+    applyStretchAlignmentToChildIfNeeded(child);
+
```

Comment:
I think this calculation should happen further up, before we do the initial layout of the child. Things that might fail by doing it this far down include layout of its contents, and handling of aspect ratios. Maybe it should somehow merge with where we handle FillAvailable width values...

(Perhaps for now the best thing is to just treat stretch the same as start, and handle it in a follow-up patch?)

Reply:
Is this similar to how for flex/grid layout we need to do a re-layout because for example, children with percent dimensions need to be changed?

In that case it probably does make sense to do it in a separate patch, where we consolidate the logic of `RenderFlexibleBox::applyStretchAlignmentToFlexItem` and `RenderGrid::applyStretchAlignmentToGridItemIfNeeded` into some common method in `RenderBlockFlow`. I took a look at it myself earlier but got a bit confused on which parts used the code in `rendering` and which used the code in `layout` ...

Reply:
`RenderBlockFlow` is responsible for block/inline layout, grid and flex are `RenderBlocks`. But I'd advise against adding grid and flex only bits to base classes (we've got enough confusion  and spaghetti like functions because of this approach) instead factor them out to helper functions (classes) if needed.